experimental/LieAlgebras/src/serialization.jl

```diff
@@ -0,0 +1,125 @@
+@register_serialization_type AbstractLieAlgebra uses_id
+
+function save_object(s::SerializerState, L::AbstractLieAlgebra)
+  save_data_dict(s) do
+    save_typed_object(s, coefficient_ring(L), :base_ring)
+    save_object(s, _struct_consts(L), :struct_consts)
+    save_object(s, symbols(L), :symbols)
+    save_attrs(s, L)
+  end
+end
+
+function load_object(s::DeserializerState, ::Type{AbstractLieAlgebra})
+  R = load_typed_object(s, :base_ring)
+  struct_consts = load_object(s, Matrix, (sparse_row_type(R), R), :struct_consts)
+  symbs = load_object(s, Vector, Symbol, :symbols)
+  L = lie_algebra(R, struct_consts, symbs; check=false)
+  load_attrs!(s, L)
+  return L
+end
+
+@register_serialization_type LinearLieAlgebra uses_id
+
+function save_object(s::SerializerState, L::LinearLieAlgebra)
+  save_data_dict(s) do
+    save_typed_object(s, coefficient_ring(L), :base_ring)
+    save_object(s, L.n, :n)
+    save_object(s, matrix_repr_basis(L), :basis)
+    save_object(s, symbols(L), :symbols)
+    save_attrs(s, L)
+  end
+end
+
+function load_object(s::DeserializerState, ::Type{LinearLieAlgebra})
+  R = load_typed_object(s, :base_ring)
+  n = load_object(s, Int, :n)
+  basis = load_object(s, Vector, (dense_matrix_type(R), matrix_space(R, n, n)), :basis)
+  symbs = load_object(s, Vector, Symbol, :symbols)
+  L = lie_algebra(R, n, basis, symbs; check=false)
+  load_attrs!(s, L)
+  return L
+end
+
+@register_serialization_type DirectSumLieAlgebra uses_id
+
+function save_object(s::SerializerState, L::DirectSumLieAlgebra)
+  save_data_dict(s) do
+    save_typed_object(s, coefficient_ring(L), :base_ring)
+    save_data_array(s, :summands) do
+      for summand in L.summands
+        ref = save_as_ref(s, summand)
+        save_object(s, ref)
+      end
+    end
+    save_attrs(s, L)
+  end
+end
+
+function load_object(s::DeserializerState, ::Type{DirectSumLieAlgebra})
+  R = load_typed_object(s, :base_ring)
+  summands = Vector{LieAlgebra{elem_type(R)}}(
+    load_array_node(s, :summands) do _
+      load_ref(s)
+    end,
+  )
+
+  L = direct_sum(R, summands)
+  load_attrs!(s, L)
+  return L
+end
+
+function save_attrs(s::SerializerState, L::LieAlgebra)
+  save_data_dict(s, :attrs) do
+    for bool_prop in (:is_abelian, :is_nilpotent, :is_perfect, :is_simple, :is_solvable)
+      if has_attribute(L, bool_prop)
+        save_object(s, get_attribute(L, bool_prop), bool_prop)
+      end
+    end
+    for symbol_prop in (:type,)
+      if has_attribute(L, symbol_prop)
+        save_object(s, get_attribute(L, symbol_prop), symbol_prop)
+      end
+    end
+    # TODO: handle root_system
```

Comment:
@HereAround @emikelsons please wait for this TODO to be resolved (ETA end of August) before putting Lie algebras inside your FTheoryTools serialization. Or, alternatively, have a script that re-creates the serialized models so that we can just re-run it once the serialization of Lie algebras changes

Reply:
Edit: I already managed to resolve this today. But you should have a reliable way to re-generate your FTheoryTools models anyway, since there certainly will be changes to the Lie algebra serialization. (Not necessarily breaking, but adding more information about them and thus e.g. speeding up some operations on them)

Reply:
Sorry, this slipped my attention.

All good. I have opened https://github.com/oscar-system/Oscar.jl/pull/4008 to coordinate this and many other serialization changes for `FTheoryTools`. In other words, waiting for you and @antonydellavecchia to tell us when/how to proceed.